parson.c

```diff
@@ -274,80 +316,103 @@ static char * read_file(const char * filename) {
+    if (size_read == 0 || ferror(fp)) {
+        fclose(fp);
+        // TODO: Shouldn't need this dynamic cast, length of >= 0 should be fine
+        parson_free(char, file_contents);
+        return NULL;
     }
     fclose(fp);
-    file_contents[file_size] = '\0';
-    return file_contents;
+    file_contents[size_read] = '\0';
+    return (_Nt_array_ptr<char>)file_contents;
 }
 
-static void remove_comments(char *string, const char *start_token, const char *end_token) {
+static void remove_comments(char* string : itype(_Nt_array_ptr<char>), _Nt_array_ptr<const char> start_token, _Nt_array_ptr<const char> end_token) {
     int in_string = 0, escaped = 0;
     size_t i;
-    char *ptr = NULL, current_char;
-    size_t start_token_len = strlen(start_token);
-    size_t end_token_len = strlen(end_token);
-    if (start_token_len == 0 || end_token_len == 0) {
-        return;
-    }
-    while ((current_char = *string) != '\0') {
-        if (current_char == '\\' && !escaped) {
-            escaped = 1;
-            string++;
-            continue;
-        } else if (current_char == '\"' && !escaped) {
-            in_string = !in_string;
-        } else if (!in_string && strncmp(string, start_token, start_token_len) == 0) {
-            for(i = 0; i < start_token_len; i++) {
-                string[i] = ' ';
-            }
-            string = string + start_token_len;
-            ptr = strstr(string, end_token);
-            if (!ptr) {
-                return;
-            }
-            for (i = 0; i < (ptr - string) + end_token_len; i++) {
-                string[i] = ' ';
+    _Unchecked {
+        char current_char;
+        size_t start_token_len = strlen(start_token);
+        size_t end_token_len = strlen(end_token);
+        if (start_token_len == 0 || end_token_len == 0) {
+            return;
+        }
+        while ((current_char = *string) != '\0') {
+            if (current_char == '\\' && !escaped) {
+                escaped = 1;
+                string++;
+                continue;
+            } else if (current_char == '\"' && !escaped) {
+                in_string = !in_string;
+            // TODO: Can't prove this
+            } else {
+                _Unchecked {
+                    if (!in_string && strncmp(string, start_token, start_token_len) == 0) {
+                        for(i = 0; i < start_token_len; i++) {
+                            string[i] = ' ';
+                        }
+                        string = string + start_token_len;
+                        char* ptr = strstr(string, end_token);
+                        if (!ptr) {
+                            return;
+                        }
+                        for (i = 0; i < (ptr - string) + end_token_len; i++) {
+                            string[i] = ' ';
+                        }
+                        // TODO: Don't understand the compiler warning here. Yes string bounds have changed to ptr, ptr + 0, what is the complaint?
```

Comment:
The problem is that `ptr` is an unchecked pointer and has no bounds.  So the compiler can't prove that `ptr + end_token_len - 1` points to a valid checked string.   We could make `ptr` a checked string, but the compiler still won't understand what `strstr` means for length (that `ptr` has bounds of `count(strlen(endtoken))`.